lib/osf-components/addon/components/editable-field/provider-metadata-manager/component.ts

```diff
@@ -0,0 +1,93 @@
+import { action } from '@ember/object';
+import { alias, and } from '@ember/object/computed';
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency-decorators';
+import DS from 'ember-data';
+import Intl from 'ember-intl/services/intl';
+import Toast from 'ember-toastr/services/toast';
+
+import Registration, { ProviderMetadata } from 'ember-osf-web/models/registration';
+import CurrentUserService from 'ember-osf-web/services/current-user';
+import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+
+export interface Args {
+    registration: Registration;
+}
+
+export interface ProviderMetadataManager {
+    save: () => void;
+    cancel: () => void;
+    inEditMode: boolean;
+    currentProviderMetadata: ProviderMetadata[];
+    userCanEdit: boolean;
+}
```

Comment:
The yielded hash exposes more fields/actions than what's defined in the interface here. Not a big deal for now, but good to take of this since we plan to type-check templates soon...